book/analysis/performance-of-python-types.md

```diff
@@ -5,90 +5,74 @@ position: 4
 layout: default.html
 ---
 
-Now that you have a general idea of big O notation and the differences
-between the different functions, our goal in this section is to tell you
-about the big O performance for the operations on Python lists and
-dictionaries. It is important for you to understand the efficiency of
-these Python data structures because they are the building blocks we
-will use as we implement other data structures in the remainder of the
-book. In this section we will try to give you some intuitive understanding
-of why the performance is what it is, but you will not fully appreciate
-why this is the case until later sections where we dive into the details
-of how lists and dictionaries may be implemented.
+Now that you have a general idea of big O notation and the differences among
+functions, our goal in this section is to tell you about the big O performance
+for each operation supported by Python lists and dictionaries. The efficiencies
+of these data types are important because we use them to implement other
+abstract data structures in the remainder of this book.
+
+This section is intended to give you some intuitive understanding of *why* the
+performances are what they are, but you will not fully appreciate these reasons
+until later, when we explore how lists and dictionaries may be implemented.
```

Comment:
I feel like we should clarify the difference between the Python language and the CPython implementation. As far as I'm aware, the language specification does not dictate the performance characteristics of an implementation, so it's possible that the information here is not necessarily correct to an implementation. If nothing else, I suggested another change below which refers to the C array underlying a Python list. Of course this is the wrong language for Jython, IronPython etc so we should resolve that.